src/sparsematrix.jl

```diff
@@ -3908,27 +3908,24 @@ function vcat(X::AbstractSparseMatrixCSC...)
         ptr_res = colptr[c]
         for i = 1 : num
             colptrXi = getcolptr(X[i])
-            col_length = (colptrXi[c + 1] - 1) - colptrXi[c]
+            rowvalXi = rowvals(X[i])
+            nzvalXi = nonzeros(X[i])
+            col_length = colptrXi[c + 1] - colptrXi[c]
             ptr_Xi = colptrXi[c]
 
-            stuffcol!(X[i], colptr, rowval, nzval,
-                      ptr_res, ptr_Xi, col_length, mX_sofar)
+            stuffcol!(rowval, nzval, ptr_res, rowvalXi, nzvalXi, ptr_Xi, col_length, mX_sofar)
 
-            ptr_res += col_length + 1
+            ptr_res += col_length
             mX_sofar += mX[i]
         end
         colptr[c + 1] = ptr_res
     end
     SparseMatrixCSC(m, n, colptr, rowval, nzval)
 end
 
-@inline function stuffcol!(Xi::AbstractSparseMatrixCSC, colptr, rowval, nzval,
-                           ptr_res, ptr_Xi, col_length, mX_sofar)
-    colptrXi = getcolptr(Xi)
-    rowvalXi = rowvals(Xi)
-    nzvalXi  = nonzeros(Xi)
-
-    for k=ptr_res:(ptr_res + col_length)
+@inline function stuffcol!(rowval, nzval, ptr_res, rowvalXi, nzvalXi, ptr_Xi,
```

Comment:
add an alias for the original method on AbstractSparsematrixCSC

Reply:
Sorry, I don't really understand what you mean by that, could you elaborate?

Reply:
It might make sense to add a method that makes any code that uses the old signature of `stuffcol!` work (and would call the new function),

you decided if it makes sense.

Reply:
Please add an overload to stuffcol! that would keep all the existing calls to that function working.

Reply:
Oh I see, thanks. Inside of `SparseArrays`, the only occurence of `stuffcol!` was really as this small helper function for `vcat`, hence why it was the only call site that had to be adapted. Or should we be concerned about external users of `SparseArrays` possibly depending on it? I can hardly imagine anyone using it, since it is not documented in any way as part of the API as far as I can tell. But I am not sure about the usual policy in Julia regarding such breakages, with technically any function being visible I suppose. So I could add back a version with the old signature if you prefer. However then that version should also still have the old somewhat confusing behavior in regard to the parameter `col_length` I guess. Seems a little ugly.

Reply:
hmmm, let's leave it as is then.